lib/internal/process/execution.js

```diff
@@ -33,26 +33,30 @@ function tryGetCwd() {
   }
 }
 
-function evalScript(name, body, breakFristLine) {
+function evalScript(name, body, breakFirstLine) {
   const CJSModule = require('internal/modules/cjs/loader');
-  if (breakFristLine) {
-    const fn = `function() {\n\n${body};\n\n}`;
-    body = `process.binding('inspector').callAndPauseOnStart(${fn}, {})`;
-  }
+  const { kVmBreakFirstLineSymbol } = require('internal/util');
 
   const cwd = tryGetCwd();
 
   const module = new CJSModule(name);
   module.filename = path.join(cwd, name);
   module.paths = CJSModule._nodeModulePaths(cwd);
-  const script = `global.__filename = ${JSON.stringify(name)};\n` +
-                  'global.exports = exports;\n' +
-                  'global.module = module;\n' +
-                  'global.__dirname = __dirname;\n' +
-                  'global.require = require;\n' +
-                  'return require("vm").runInThisContext(' +
-                  `${JSON.stringify(body)}, { filename: ` +
-                  `${JSON.stringify(name)}, displayErrors: true });\n`;
+  global.kVmBreakFirstLineSymbol = kVmBreakFirstLineSymbol;
+  const script = `
+    global.__filename = ${JSON.stringify(name)};
+    global.exports = exports;
+    global.module = module;
+    global.__dirname = __dirname;
+    global.require = require;
+    const { kVmBreakFirstLineSymbol } = global;
+    delete global.kVmBreakFirstLineSymbol;
+    return require("vm").runInThisContext(
+      ${JSON.stringify(body)}, {
+        filename: ${JSON.stringify(name)},
+        displayErrors: true,
+        [kVmBreakFirstLineSymbol]: ${!!breakFirstLine}
+      });\n`;
```

Comment:
@addaleax The `kVmBreakFirstLineSymbol` looks pretty handy. Is there a reason to keep it hidden and not exposed as a user facing option like a boolean `breakFirstLineSymbol` options property?

Reply:
@jdalton Err, it’s less work for this PR? :smile: I’m not sure, but I guess it also only works while the inspector is enabled, similar to `debugger;` statements?

Somebody from @nodejs/v8-inspector might know more.